engine/table/src/main/java/io/deephaven/engine/table/impl/lang/QueryLanguageParser.java

```diff
@@ -846,23 +848,27 @@ private static int costToAssignFrom(final Class<?> dest, final Class<?> src) {
             return CandidateExecutable.ALREADY_EQUAL;
         }
 
-        if ((dest.isPrimitive() && dest != boolean.class) && src.isPrimitive() && src != boolean.class) {
+        if (dest.isPrimitive() && dest != boolean.class && src.isPrimitive() && src != boolean.class) {
             if (dest == binaryNumericPromotionType(dest, src)) {
                 return CandidateExecutable.WIDEN_CONVERSION;
             }
             return CandidateExecutable.NO_CONVERSION_POSSIBLE;
-        } else if (!dest.isPrimitive() && src == NULL_CLASS) {
-            return CandidateExecutable.ALREADY_EQUAL;
         }
 
-        Class<?> boxedDest = io.deephaven.util.type.TypeUtils.getBoxedType(dest);
-        Class<?> boxedSrc = io.deephaven.util.type.TypeUtils.getBoxedType(src);
+        if (src == NULL_CLASS) {
```

Comment:
Should this be an unboxing conversion for non-boolean? I guess that's a question of what other code does, downstream.